File: gitnexus/test/fixtures/lang-resolution/csharp-overload-param-types/Models/UserService.cs
namespace Models;

public class User
{
    public string GetName() => "user";
}

public class UserService
{
    public User Lookup(int id)
    {
        return new User();
    }

    public User Lookup(string name)
    {
        return new User();
    }

    public void Run()
    {
        Lookup(42);        // literal int → should disambiguate to Lookup(int)
        Lookup("alice");   // literal string → should disambiguate to Lookup(string)
    }
}

File: gitnexus/test/fixtures/lang-resolution/java-overload-param-types/models/UserService.java
package models;

public class UserService {
    public User lookup(int id) {
        return new User();
    }

    public User lookup(String name) {
        return new User();
    }

    public void run() {
        lookup(42);        // literal int → should disambiguate to lookup(int)
        lookup("alice");   // literal String → should disambiguate to lookup(String)
    }
}
